### ISSUE TITLE
Get limit exceeded

### ISSUE THREAD

**Issue by it-slav** [py-smart-gardena/py-smart-gardena#79]
Hi

I am new to python so bare with me.

I want to get status of my Gardena Mover sent to mqtt to be handled by node-red.

If I read the documentation correctly I should be able to read the values of my Gardena Mover as often as I want. The values will be updated via websocket only when the status change.
However if I read the values too often like every 10 sec, I get an error after a day or so "ERROR    429 : Limit Exceeded".
I guess the reason is that hit the 10k/monthA API limit. Is the documentation wrong or do I understand it incorrectly, or something else?

Any hints how I can make it only read the values when any value change would be appreciated, I have tried to fiddle with callback but could never get that working. 

Thanks
PeterA

```
from gardena.smart_system import SmartSystem
import pprint
import time
import json
import paho.mqtt.publish as mqttpublish

smart_system = SmartSystem(email="<email>", password="<password>", client_id="<client-id>")
smart_system.authenticate()


while 1:
  print ("")
  localtime = time.localtime()
  result = time.strftime("%I:%M:%S %p", localtime)
  print(result)
  smart_system.update_locations()
  for location in smart_system.locations.values():
      smart_system.update_devices(location)
      pprint.pprint(location)
      for device in location.devices.values():
          pprint.pprint(device)
  for location in smart_system.locations.values():
    print("location : " + location.name + "(" + location.id + ")")
  for device in smart_system.locations[location.id].find_device_by_type("MOWER"):
    print(f"name : {device.name}")
    print(f"id : {device.id}")
    print(f"type : {device.type}")
    print(f"model_type : {device.model_type}")
    print(f"battery_level : {device.battery_level}")
    print(f"battery_state : {device.battery_state}")
    print(f"rf_link_level : {device.rf_link_level}")
    print(f"rf_link_state : {device.rf_link_state}")
    print(f"serial : {device.serial}")
    print(f"activity : {device.activity}")
    print(f"operating_hours : {device.operating_hours}")
    print(f"state : {device.state}")
    print(f"last_error_code : {device.last_error_code}")
    msg=json.dumps({'name': device.name, 'battery_level': device.battery_level, 'battery_state': device.battery_state, 'rf_link_level': device.rf_link_level, 'rf_link_state': device.rf_link_state, 'activity': device.activity, 'operating_hours': device.operating_hours, 'state': device.state, 'last_error_code': device.last_error_code})
    print("Sending:" + msg)
    mqttpublish.single("gardena/mover", msg, hostname="localhost")

  time.sleep (10)
```

**Comment by grm** [comment]
indeed, there s a rate limit and a maximum numer of calls per month.
You should use websocker communication as the websocket gets notified of every change as soon as they arrive, so you don't need to poll their API !

Hope it helps,
Jérémie

**Comment by it-slav** [comment]
That was not an answer to my questions.

1. "If I read the documentation correctly I should be able to read the values of my Gardena Mover as often as I want. The values will be updated via websocket only when the status change."
Correct?

Cut-and-paste from doc:
--start--
Devices are automatically retrieved the first time from the API, and then the websocket is used to get updates. They are stored in each locations. Depending on the function type, you can have diffrents fields.
--end--
I interpret this as the the values are ONLY updated when status has changed, so by looping it like I do will not generate and API call to Husqvarna. This seems like it  is not true as I hit the limit?

2. Do I understand the doc wrong?

Thanks
PeterA

**Comment by Domochip** [comment]
I developped a Docker Image that send Gardena infos to MQTT.
It may interests you : 
https://github.com/Domochip/gardena2mqtt
https://hub.docker.com/r/domochip/gardena2mqtt

**Comment by Domochip** [comment]
The gardena documentation explains that, once you are connected to the API, you will get updates through websocket. It means that API will send you update when there is, and Not you request it...

Could you give me the compose file you used?

**Comment by grm** [comment]
The problem with your code is that you are requesting without waiting continuously gardena api.
You should put a `sleep` in your `while(1) : `

The limit exceeded is sent by the gardena api because it detects that you are hammering too hard their API (because you doo not sleep and update coninuously).
Regarding this error, there is nothing the library can do about it ;)
